### PR TITLE
mailutils: 2.2 -> 3.2

### LINKED ISSUE #27475: mailutils fails to build on master
## Issue description
The package mailutils fails to build on master for some time now: https://hydra.nixos.org/build/56249845
I was able to reproduce the error on my machine. Git bisect lead me to commit d07f30f628f52186aa68b0c918c88aa9ed36f988 as first bad one but I do not know why it would cause the failure.

ping @ryantrinkle (author of suspected commit), @vrthra (maintainer of mailutils)

### Steps to reproduce
With current nixpkgs master:
```nix-build '<nixpkgs>' --option build-use-chroot true --option build-use-sandbox true --keep-going -A mailutils```

## Technical details
https://nix-cache.s3.amazonaws.com/log/sxc52lb5c1gd9idldqq0drz3nln65xap-mailutils-2.2.drv

* Nix version: 1.11.11

### PR DESCRIPTION
**Pull request by orivej**
###### Motivation for this change

Fixes #27475. (Probably leaves #21008 open.)

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/mailutils/default.nix [hunk 1]
```diff
@@ -1,36 +1,58 @@
...
+    "--with-mysql"
   ];
 
-  buildInputs =
-   [ gettext gdbm libtool pam readline ncurses
-     gnutls mysql.lib guile texinfo gnum4 sasl fribidi gss nettools ]
-   ++ stdenv.lib.optional doCheck dejagnu;
+  preCheck = ''
+    # Add missing files.
+    cp ${./readmsg-tests}/* readmsg/tests/
```
Comment (Mic92) [inline pkgs/tools/networking/mailutils/default.nix]: Where these files added later to upstream?

Comment (orivej) [inline pkgs/tools/networking/mailutils/default.nix]: They have always been in the upstream repository, but they are missing from the released archive! I copied this fix from http://data.gpo.zugaina.org/gentoo/net-mail/mailutils/mailutils-3.2.ebuild

Comment (orivej) [inline pkgs/tools/networking/mailutils/default.nix]: We could use repository snapshot which contains these files: http://git.savannah.gnu.org/cgit/mailutils.git/snapshot/mailutils-release-3.2.tar.gz

Comment (Mic92) [inline pkgs/tools/networking/mailutils/default.nix]: Ok, then please switch to the other url.

Comment (orivej) [inline pkgs/tools/networking/mailutils/default.nix]: It turned out to be more involved. Bootstrapping the build requires a git checkout (and wget, and rsync), so I used fetchgit and added additional dependencies to run `./bootstrap`.

**Comment by globin** [comment]
This fails with sandboxing turned on. I'll have a look if I can fix this right away.

@Mic92 the patch didn't apply, I added an `excludes` param to `fetchpatch`.

**Comment by orivej** [comment]
I tested with sandboxing.

A travis job suddenly failed with `fetchgit` hash mismatch: https://travis-ci.org/NixOS/nixpkgs/jobs/256661069 . It used to pass, and `fetchgit` arguments have not changed. I don't know how this is possible.

File: pkgs/tools/networking/mailutils/default.nix [hunk 2]
```diff
@@ -29,7 +29,8 @@ stdenv.mkDerivation rec {
   patches = [
     (fetchpatch {
       url = "http://git.gnu.org.ua/cgit/mailutils.git/patch/?id=afbb33cf9ff750e93a9a4c1f51a3b62d584f056e";
-      sha256 = "1w72ymxlkqj2y3gqd2r27g79hxw7xa6j790shsg9i5jhhz3vknjx";
+      sha256 = "0yzkfx3j1zkkb43fhchjqphw4xznbclj39bjzjggv32gppy6d1db";
```
Comment (Mic92) [inline pkgs/tools/networking/mailutils/default.nix]: Sorry I only compared commit hash.

**Comment by orivej** [comment]
It seems that making git checkout work is too much effort. I'd prefer to return to the first commit in this pull request, but download the patch and missing files from gentoo repository: #27595

**Comment by clefru** [comment]
FYI this builds fine for me locally. (Don't have much more to test than building unfortunately). Thanks for the fix, orivej!

**Comment by globin** [comment]
Superseeded by #27595

**Pull request closed without merging**